### PR TITLE
Fixed #68 - Implemented submodule & gh-pages update task, "publish"

### LINKED ISSUE #68: Fix the brackets development workflow
This is a couple of things:

1.  Get rid of as many submodules as possible
2.  Add `grunt` tasks for:
  1. Automatically updating our submodules/packages for extensions
  2. Automatically build and deploy our latest code to heroku

### PR DESCRIPTION
**Pull request by sedge**
The only piece that isn't here is documentation. Should I put it in the README?

### CODE REVIEW

File: tasks/test.js [hunk 1]
```diff
@@ -1,24 +1,24 @@
 /*
  * Copyright (c) 2013 Adobe Systems Incorporated. All rights reserved.
- *
```
Comment (sedge) [inline tasks/test.js]: ugh

File: src/config.json [hunk 2]
```diff
@@ -36,29 +36,33 @@
     },
     "devDependencies": {
         "grunt": "0.4.1",
-        "jasmine-node": "1.11.0",
-        "grunt-jasmine-node": "0.1.0",
         "grunt-cli": "0.1.9",
-        "phantomjs": "1.9.11",
```
Comment (sedge) [inline src/config.json]: Hmm, not sure what happened here...

Comment (sedge) [inline src/config.json]: I bet this is a rebase error - I never touched this file.

File: package.json [hunk 3]
```diff
@@ -1,52 +1,57 @@
 {
-    "name": "Brackets",
```
Comment (sedge) [inline package.json]: I'll fix this.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
```
Comment (sedge) [inline index.html]: This wasn't supposed to be committed.

File: Gruntfile.js [hunk 5]
```diff
@@ -343,9 +361,125 @@ module.exports = function (grunt) {
...
+                    // clarity. Their actual values
+                    // will be set by the `publish` task.
+                    branch: 'gh-pages',
+                    force: true
+                },
+            }
         }
     });
 
+    // Bramble-task: smartCheckout
```
Comment (humphd) [inline Gruntfile.js]: Can you document these a bit more than just calling them "smart"?

**Comment by humphd** [comment]
Add some more helpful comments, and this looks good to me.

**Pull request merged**